domain/cache/caching_repo_request_driven.py

```diff
@@ -132,6 +133,7 @@ def _update_cache_with(self, results: MeshResults, config: MeshConfig) -> None:
             logger.debug("Incremental cache update")
             new_results = deepcopy(current_results)
             new_results.incremental_update(results)
+            self._drop_samples_outside_timewindow(new_results)
             new_config = current_config
         else:
             logger.debug("New mesh test configuration detected. Full cache update")
```

Comment:
It should be possible to preserve accumulated history for agents that remained in the test configuration. Incremental update should use the new agent config and adjust rows and columns to match new set of agents. Worst case we end-up with a completely new matrix :).

This can be implemented in a separate PR.

Reply:
Yes it should be possible. I can do it if you find it worthwhile